### DISCUSSION TITLE
Nuxt SPA instead of client-server

### DISCUSSION THREAD

**Discussion by Glandos** [mealie-recipes/mealie#2406] [General]
Hi!

Since it's my first time here: thanks for the great work on this software!

I'm self-hosting a lot of software on my personal server at home, which has a low-end CPU, and only 4GB RAM. On top of that, my philosophy is to use as less resources as possible, so I am investigating to avoid Node.js for the backend, since there is already a Python server.

My question is: is there any real difference between `yarn build --spa` and `yarn build` (which launch `nuxt build` with the default `client + server` setting, requiring Node.js afterwards)?

As far as I've tested, the `--spa` flags generated a bunch of static files, that make the application working well with the appropriate web server settings. But maybe I've missed something? I've seen the comment in https://github.com/hay-kot/mealie/discussions/1073#discussioncomment-2995452 but I still don't see the real value (or uses) of "Server-side rendering" and "Generated Social cards".

**Comment by hay-kot** [comment]
Social Cards and improved client performance is the only real reason we have SSR and the Node server.

If you don't care about those, Spa mode should work fine.
